solrmonitor/zkdispatcher.go

```diff
@@ -24,24 +24,49 @@ import (
 	"github.com/samuel/go-zookeeper/zk"
 )
 
-// Handle a zk event.  Optionally return a new event channel to automatically re-register.
+// ugh... seems silly to have to define this
+const maxInt = int(^uint(0) >> 1)
+
+// A function that handles a zk event. Optionally returns a new event channel to automatically
+// re-register.
 type ZkEventHandler func(zk.Event) <-chan zk.Event
 
-// Monitors many zk.Event channels on a single goroutine.
+// An interface that handles a zk event. Optionally returns a new event channel to automatically
+// re-register.
+type ZkEventCallback interface {
+	Handle(zk.Event) <-chan zk.Event
+}
+
+// Adapts a ZkEventHandler function to the ZkEventCallback interface.
+type zkEventHandlerAdapter struct {
+	handler ZkEventHandler
+}
+
+func (a *zkEventHandlerAdapter) Handle(event zk.Event) <-chan zk.Event {
+	return a.handler(event)
+}
```

Comment:
by the way, you may already know this, but there's a neat trick to convert a mono func into an interface...

```
func (f ZkEventHandler) Handle(event zk.Event) <-chan zk.Event {
	return f(event)
}
```

Unfortunately it doesn't work here because you end up using the adapter as a map key.  Probably want to add a comment that we need the struct to be a key.